### PR TITLE
docs: add document for idl_trimmer tool

### PR DESCRIPTION
**Pull request by tksky1**
#### What type of PR is this?
docs

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en: add document for thriftgo-idl-trimmer tool
zh(optional):

### CODE REVIEW

File: static/img/docs/idl_trimmer_process2.jpg
Comment (HeyJavaBean) [inline static/img/docs/idl_trimmer_process2.jpg:1]: 你图片有水印....

**Review by HeyJavaBean — APPROVED** [review]

File: static/img/docs/idl_trimmer_process2.jpg
Comment (HeyJavaBean) [reply static/img/docs/idl_trimmer_process2.jpg:1]: 我ap点错了...忽略下....

**Review by rogerogers — CHANGES_REQUESTED** [review]
图片水印

File: static/img/docs/idl_trimmer_process2.jpg
Comment (tksky1) [reply static/img/docs/idl_trimmer_process2.jpg:1]: fixed

File: content/zh/docs/kitex/Tutorials/code-gen/idl_trimmer.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+title: "Thriftgo IDL 裁切工具"
+date: 2023-11-08
+weight: 8
+keywords: ["trimmer", "IDL裁切", "IDL裁剪"]
+description: "使用IDL裁切工具裁切不必要的IDL定义"
+---
+
+## 简介
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/code-gen/idl_trimmer.md:9]: 简介开头建议加一个 thrifgo 的简介和超链接。

File: content/zh/docs/kitex/Tutorials/code-gen/idl_trimmer.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+在Thriftgo v0.3.3 版本（暂未发布，拉取最新commit即可）中，指定method裁切功能支持了正则表达式的method匹配方式。用户可以构造正则表达式来匹配`[serviceName].[methodName]`的方法名，精确指定一个或多个方法。例如：  
+```bash
+trimmer -m 'Employee.*\..*' test_cases/sample1.thrift`  
+```
+可以匹配“Employee”开头的service下的全部方法。（注：“.”在正则表达式有意义，最好使用\.来匹配）  
+也可以使用一些perl风格的表达式进行一些高级裁切操作，例如：  
+```bash
+trimmer -m '^(?!EmployeeService.getEmployee).*$' test_cases/sample1.thrift
+```
+可以匹配除了EmployeeService.getEmployee外的所有方法。执行这条命令，可以单独从IDL中删除该方法及其依赖（且其他方法无依赖）的结构。
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/code-gen/idl_trimmer.md:101]: 本文中英文间需要加一个空格，建议统一处理下

File: content/zh/docs/kitex/Tutorials/code-gen/idl_trimmer.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+以 KiteX 为例，使用时添加 `-thrift trim_idl` 参数，例如：  
+```bash
+`kitex -module xx -thrift trim_idl xxx.thrift`
+```
+使用该参数时，命令行会额外输出额外提示：  
+`[WARN] You Are Using IDL Trimmer`  
+代码生成时，将会对所用到 IDL 进行裁切，最终生成的 Golang 代码就不会有无用的 Struct 结构体了。
+
+## 使用 yaml 进行配置
+为了方便通过 KiteX/Thriftgo 集成方法进行裁切时使用 trimmer tool 的参数配置，Thriftgo v0.3.3 版本（暂未发布，拉取最新commit即可）为 trimmer 工具提供了 yaml 配置文件支持。在集成 KiteX/Thriftgo 或直接使用 trimmer 工具时，trimmer 都会自动扫描并应用当前执行的目录（os.Getwd()）下的 trim_config.yaml 配置文件。
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/code-gen/idl_trimmer.md:130]: KiteX -> Kitex 全文替换

**Comment by GuangmingLuo** [comment]
补一个英文文档吧

**Comment by tksky1** [comment]
> 补一个英文文档吧

搞定了

**Review by HeyJavaBean — APPROVED** [review]

**Review by GuangmingLuo — APPROVED** [review]

**Pull request merged**